sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/AsyncResetTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import static io.opentelemetry.sdk.testing.assertj.metrics.MetricAssertions.assertThat;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.ObservableLongMeasurement;
+import io.opentelemetry.sdk.metrics.testing.InMemoryMetricReader;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+
+class AsyncResetTest {
+
+  @Test
+  void demonstrateAsyncCumulativeRetentionBug() {
```

Comment:
This test is redundant to `TemporalMetricStorageTest#asynchronous_CumulativeRetainsHistory`, but I think it helps illustrate the point better.

I think we should remove it before merging.